nodejs/eks/cluster.ts

```diff
@@ -242,8 +242,8 @@ export function generateKubeconfig(
 }
 
 export interface ClusterCreationRoleProviderOptions {
-    region?: aws.Region;
-    profile?: string;
+    region?: pulumi.Input<aws.Region>;
+    profile?: pulumi.Input<string>;
```

Comment:
These are non-plain `Input<T>` in the schema:

https://github.com/pulumi/pulumi-eks/blob/c6f1fce98fa2683abe7ffbad7f166da99928fb83/provider/cmd/pulumi-gen-eks/main.go#L619-L624

And they can easily be made `Input<T>` in the implementation, as they're just passed along through `getRoleProvider`, which passes them as inputs when creating `aws.Provider`, which already accepts `Input<T>` for these:

https://github.com/pulumi/pulumi-eks/blob/c6f1fce98fa2683abe7ffbad7f166da99928fb83/nodejs/eks/cluster.ts#L331-L333